opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -298,17 +304,21 @@ namespace Opm {
                 updateState(x,reservoir_state);
                 wellModel().updateWellState(xw, well_state);
 
-                // since the solution was changed, the cache for the intensive quantities
-                // are invalid
-                ebosSimulator_.model().invalidateIntensiveQuantitiesCache(/*timeIdx=*/0);
+                // since the solution was changed, the cache for the intensive quantities are invalid
+                // ebosSimulator_.model().invalidateIntensiveQuantitiesCache(/*timeIdx=*/0);
+            }
 
-                // solver has succeed i.e. no need for restart.
-                isRestart_ = false;
+            if( converged && (iteration >= nonlinear_solver.minIter()) )
+            {
+                // in case of convergence we do not need to reset intensive quantities
```

Comment:
"converged" here referees to the non-linear solver. If the non-linear solver has converged you don't need to solve the linear system and update the solution.  The problem with the current approach is that it doesn't recalculate the intensiveQs when the linear solver doesn't converge.